CMSgov/easi-app
# [Easi-2789] - Replace CEDAR LDAP with Okta

# EASI-2789

## Changes and Description

- Replaces the existing CEDAR User Search API with a generic search API using Okta

AC also calls for a cleanup ticket which can be [found here](https://jiraent.cms.gov/browse/EASI-3028).

## How to test this change

The following steps should fire all three functions in the `userSearch` interface: `FetchUserInfo`, `FetchUserInfos`, and `SearchCommonNameContains`. All of the tests for the app mock the current Cedar Ldap Client interface, rendering them useless for testing the changes, unfortunately.

1. [Change this if statement to false to force the Okta API locally.](https://github.com/CMSgov/easi-app/blob/EASI-2789/cedar_ldap_to_okta/pkg/server/routes.go#L107)
2. Add the following to your `.envrc.local`:

```
export OKTA_API_URL="https://test.idp.idm.cms.gov"
export OKTA_API_TOKEN="<okta-token-from-1pass>"
```
3. Start the app with `scripts/dev up`
4. [Login to the app](http://localhost:3000) with your EUA.
5. Create a TRB Request
6. ~~[Comment out this code to see the TRB Admin dashboard.](https://github.com/CMSgov/easi-app/blob/EASI-2789/cedar_ldap_to_okta/src/views/Home/index.tsx#L67-L86)~~ There's now a dash switcher component in main.
7. Add yourself (or someone else) to TRB Lead options through the [GraphQL Playground](http://localhost:8080/api/graph/playground):

```gql
mutation($euaId: String!) {
    createTrbLeadOption(eua: $euaId) {
       euaUserId
       email
    }
}
```

8. Verify you've added a TRB Lead option:

```gql
query {
    trbLeadOptions {
        euaUserId
        email
        commonName
    }
}
```
9. Attach that person to the TRB Request as a lead through the Admin dashboard UI (may require a refresh).
10. Find Someone by Common Name (first/last):

```gql
query($commonName: String!) {
    cedarPersonsByCommonName(commonName: $commonName) {
        commonName
        email
        euaUserId
    }
}
```

## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
